backend/app/services/chat/preprocessing/contexts.py

```
@@ -0,0 +1,659 @@
+
+            # Process based on context type
+            if context.context_type == ContextType.ATTACHMENT.value:
+                _process_attachment_context(context, idx, text_contents, image_contents)
+            elif context.context_type == ContextType.KNOWLEDGE_BASE.value:
+                # Knowledge base contexts are handled via RAG tools, not here
+                logger.debug(
+                    f"Knowledge base context {context_id} will be used via RAG"
+                )
+
+        except (ValueError, KeyError) as e:
+            logger.exception(f"Error processing context {context_id}: {e}")
```

Ruff (0.14.10)

Redundant exception object included in `logging.exception` call

(TRY401)